### PR TITLE
Peer authentication sender

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added peer authentication sender
- added routineHandler & timerWrapper sub-components
- renamed some interfaces, stub implementations and errors

### CODE REVIEW

File: heartbeat/sender/peerAuthenticationSender_test.go [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+		sender, _ := NewPeerAuthenticationSender(args)
+
+		err := sender.execute()
+		assert.Nil(t, err)
+
+		skBytes, _ := sender.privKey.ToByteArray()
+		pkBytes, _ := sender.publicKey.ToByteArray()
+		log.Info("args", "pid", args.Messenger.ID().Pretty(), "bls sk", skBytes, "bls pk", pkBytes)
+
+		// verify the received bytes if they can be converted in a valid peer authentication message
```
Comment (dragos-rebegea) [inline heartbeat/sender/peerAuthenticationSender_test.go:300]: you can transform these comments into t.Run("description")

**Review by dragos-rebegea — APPROVED** [review]

File: heartbeat/sender/peerAuthenticationSender_test.go (hunk 1 above)
Comment (iulianpascalau) [reply heartbeat/sender/peerAuthenticationSender_test.go:300]: can be done, sure

**Review by sstanculeanu — DISMISSED** [review]

File: heartbeat/sender/peerAuthenticationSender.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-go/heartbeat"
+)
+
+const minTimeBetweenSends = time.Second
+
+// ArgPeerAuthenticationSender represents the arguments for the heartbeat sender
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply heartbeat/sender/peerAuthenticationSender.go]: done

File: heartbeat/sender/peerAuthenticationSender_test.go (hunk 1 above)
Comment (iulianpascalau) [reply heartbeat/sender/peerAuthenticationSender_test.go:300]: done

**Review by sstanculeanu — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**